### PR TITLE
feat!: replace `Binary` for `Uint8Array`

### PR DESCRIPTION
**Pull request by voliva**
The `Binary` class has given us some problems in v1, specially on projects that have a dependency to substrate-connect, as it creates duplicate `Binary` class definitions that are incompatible with each other.

This PR removes the `Binary` class and goes back to using `Uint8Array`s, the JS primitive for binary data. Then `Binary` becomes just a set of utility functions to work with that.

The migration path for this is simple enough. In many cases, I've found that it's even compatible, since our codecs already accepted different binary formats. And for the others, TS will complain, and it's a matter of just using the utility instead of the class-based methods.

### CODE REVIEW

File: packages/client/reexports/utils.ts [hunk 1]
```diff
@@ -37,9 +37,6 @@ export const jsonDeserialize: (this: any, key: string, value: any) => any = (
   if (typeof v === "string") {
     // this has no conflicts in papi objects.
     if (/^[0-9]+n$/.test(v)) v = BigInt(v.slice(0, -1))
-    // only lowercase and with 0x, as it comes from `toHex`
-    // this has a corner-case conflict with an AccountId20 with all lowercase checksum
-    if (/^0x([0-9a-f][0-9a-f])+$/.test(v)) v = Binary.fromHex(v)
```
Comment (carlosala) [inline packages/client/reexports/utils.ts:42]: This could be a problem. `serialize(deserialize(obj))` will now turn `Uint8Array` into `HexString`. Maybe we can leave a marker or something to show that this is a variable-length array.

Comment (voliva) [reply packages/client/reexports/utils.ts:42]: Yeah, I thought about this one when I removed these lines but... no. I'd rather drop the assumption it's going to be the exact inverted result rather than hack around.

Problem is fixed-size binaries are represented as hex and should be kept as hex when deserialising.... and there's no standard marker for differentiating these two.

And if someone needs that guarantee, they are always free to add their own marker with an enhancer.

File: packages/observable-client/src/chainHead/track-tx.ts [hunk 2]
```diff
@@ -157,3 +157,5 @@ export const getTrackTx = (
       ),
     )
 }
+
+const u8eq = (a: Uint8Array, b: Uint8Array) => a.every((v, i) => b[i] === v)
```
Comment (carlosala) [inline packages/observable-client/src/chainHead/track-tx.ts]: smol nit
```suggestion
const u8eq = (a: Uint8Array, b: Uint8Array) =>
  a.length === b.length && a.every((v, i) => b[i] === v)
```

File: packages/signers/meta-signers/src/multisig-signer.ts [hunk 3]
```diff
@@ -155,7 +151,7 @@ export function getMultisigSigner<Address extends SS58String | HexString>(
                 call: callCodec.dec(callData),
               }
             : {
-                call_hash: Binary.fromBytes(callHash),
+                call_hash: callHash,
```
Comment (carlosala) [inline packages/signers/meta-signers/src/multisig-signer.ts]: It is a fixed-sized array. I'll be hex from now on.
```suggestion
                call_hash: toHex(callHash),
```

**Review by carlosala — COMMENTED** [review]
suuper nice 🚀 

just some quick comments

File: packages/observable-client/src/chainHead/track-tx.ts (hunk 2 above)
Comment (voliva) [reply packages/observable-client/src/chainHead/track-tx.ts]: it's ackchyually an important one

Comment (carlosala) [reply packages/observable-client/src/chainHead/track-tx.ts]: yep, internal security audit uorkss
